source/mainnet/smart-contracts/tutorials/counter/counter-contract.rst

```diff
@@ -25,215 +25,211 @@ Once you have set up the tools needed you are ready to create your smart contrac
+        let input: i8 = ctx.parameter_cursor().get()?;
 
-.. code-block:: rust
-
-    #[receive(
-        contract = "counter",
-        name = "decrement",
-        parameter = "IncrementVal",
-        error = "Error",
-        mutable
-    )]
-    fn decrement(
-        ctx: &ReceiveContext,
-        host: &mut Host<State>,
-    ) -> Result<(), Error> {
-        // Your code
-
-        let param: IncrementVal = ctx.parameter_cursor().get()?;
         let state = host.state_mut();
-        ensure!(
-            ctx.sender().matches_account(&ctx.owner()),
-            Error::OwnerError
-        );
+        let Some(result) = state.counter.checked_add(input) else {
+            return Err(Error::OverflowError);
+        };
 
-        ensure!(param < 0, Error::DecrementError);
-        state.counter += param;
+        state.counter = result;
         Ok(())
     }
 
 View function
 -------------
 
-The view function will return only the counters value so you need to update its return value as i8 and return it from the host.state().
+The view function will return only the counter's value so you need to update its return value as ``i8`` and return it from the ``host.state()``.
 
 .. code-block:: rust
 
-    /// View function that returns the content of the state.
+    /// Returns the state of the smart contract.
     #[receive(contract = "counter", name = "view", return_value = "i8")]
-    fn view<'a, 'b>(
-        _ctx: &'a ReceiveContext,
-        host: &'b Host<State>,
-    ) -> ReceiveResult<i8> {
+    fn view(_ctx: &ReceiveContext, host: &Host<State>) -> ReceiveResult<i8> {
         Ok(host.state().counter)
     }
 
 Build, deploy, and initialize the contract
 ==========================================
 
-Create a ``dist`` folder to keep the schema output file and Wasm compiled contract in and run the build command.
+Create a ``dist`` folder for the compiled WASM contract. Then, run the build command.
 
 .. code-block:: console
 
-    $cargo concordium build --out dist/module.wasm.v1 --schema-out dist/schema.bin
+    $cargo concordium build --out dist/module.wasm.v1 --schema-embed
+
+You may get a warning about the build not being verifiable, which you may ignore.
+
+Now we can deploy the smart contract using the Concordium client CLI. If you are running your own node, you can use this command:
 
-.. image:: images/build.png
-    :width: 100%
+.. code-block:: console
+
+    $concordium-client module deploy dist/module.wasm.v1 \
+        --sender <YOUR-ADDRESS> \
+        --name counter \
+        --grpc-port 20001
 
-Deploy it with the command below.
+Or, if you just want to try things out on testnet, you can use the testing nodes provided by Concordium:
 
 .. code-block:: console
 
-    $concordium-client module deploy dist/module.wasm.v1 --sender <YOUR-ACCOUNT> --name counter --grpc-port 20001
+    $concordium-client module deploy dist/module.wasm.v1 \
+        --sender <YOUR-TESTNET-ADDRESS> \
+        --name counter \
```

Comment:
I'm not sure what you're saying, do you mean naming the `module.wasm.v1` file? Or what are we setting a name for so we don't need the name parameter here?

Reply:
Yes, I don't think we need the ``--name`` option here (it can be removed since the name is not used anywhere in the tutorial).

If the ``--name MY_NAME`` option is set in the above command, you could use it:

```concordium-client contract init MY_NAME``` (no module reference needed)